File: 134-gas-station/gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {

        int n = gas.size();
        vector<int> diff(n);
        int gasSum = accumulate(gas.begin(), gas.end(), 0);
        int costSum = accumulate(cost.begin(), cost.end(), 0);

        int total = 0;
        int start = 0;

        if (gasSum < costSum) {
            return -1;
        } else {
            for (int i = 0; i < n; i++) {
                diff[i] = gas[i] - cost[i];
            }

            for (int i = 0; i < n; i++) {
                total += diff[i];
                if (total < 0) {
                    total = 0;
                    start = i + 1;
                }
            }
        }
        return start;
    }
};

File: 134-gas-station/gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        gas_sum = sum(gas)
        cost_sum = sum(cost)
        total=0
        start=0
        if(gas_sum < cost_sum):
                return -1
        for i in range(len(gas)):
            total += gas[i] - cost[i]

            if(total<0):
                total=0
                start = i+1
        
        return start
        